location/src/main/java/org/motechproject/nms/location/domain/Taluka.java

```diff
@@ -0,0 +1,120 @@
+package org.motechproject.nms.location.domain;
+
+import org.motechproject.mds.annotations.Cascade;
+import org.motechproject.mds.annotations.Entity;
+import org.motechproject.mds.annotations.Field;
+import org.motechproject.mds.annotations.UIDisplayable;
+import org.motechproject.mds.domain.MdsEntity;
+
+import javax.jdo.annotations.Persistent;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity(tableName = "nms_talukas")
+public class Taluka extends MdsEntity {
+
+    @Field
+    @UIDisplayable(position = 0)
+    @NotNull
+    private String name;
+
+    @Field
+    @UIDisplayable(position = 1)
+    @NotNull
+    private Long code;
+
+    @Field
+    @UIDisplayable(position = 2)
+    @NotNull
+    private District district;
+
+    @Field
+    @Cascade(delete = true)
+    @Persistent(mappedBy = "taluka", defaultFetchGroup = "true")
+    private List<HealthBlock> healthBlocks;
```

Comment:
Kosh and I decided that Set is hard to work with.  You can't just .get()
from it.

On Mon, Apr 27, 2015 at 5:34 PM, Frank Huster <notifications@github.com>
wrote:

> In
> location/src/main/java/org/motechproject/nms/location/domain/Taluka.java
> <https://github.com/motech-implementations/mim/pull/192#discussion_r29205814>
> :
>
> > +    private String name;
> > +
> > +    @Field
> > +    @UIDisplayable(position = 1)
> > +    @NotNull
> > +    private Long code;
> > +
> > +    @Field
> > +    @UIDisplayable(position = 2)
> > +    @NotNull
> > +    private District district;
> > +
> > +    @Field
> > +    @Cascade(delete = true)
> > +    @Persistent(mappedBy = "taluka", defaultFetchGroup = "true")
> > +    private List<HealthBlock> healthBlocks;
>
> Collections of children entities are not Set anymore?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech-implementations/mim/pull/192/files#r29205814>.
>